src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -311,25 +311,42 @@ void DofManager::addField( string const & fieldName,
 {
   GEOSX_ASSERT_MSG( m_domain != nullptr, "Domain has not been set" );
   GEOSX_ERROR_IF( m_reordered, "Cannot add fields after reorderByRank() has been called." );
-  GEOSX_ERROR_IF( fieldExists( fieldName ), "Requested field name '" << fieldName << "' already exists." );
   GEOSX_ERROR_IF_GT_MSG( components, MAX_COMP, "Number of components limit exceeded" );
 
-  m_fields.emplace_back();
-  FieldDescription & field = m_fields.back();
+  std::vector< Regions > processedRegions = processFieldRegionList( *m_domain, regions );
 
-  // populate basic info
-  field.name = fieldName;
-  field.location = location;
-  field.numComponents = components;
-  field.key = m_name + '_' + fieldName + "_dofIndex";
-  field.docstring = fieldName + " DoF indices";
-
-  // advanced processing
-  field.support = processFieldRegionList( *m_domain, regions );
-  computeFieldDimensions( static_cast< localIndex >( m_fields.size() ) - 1 );
-
-  // allocate and fill index array
-  createIndexArray( field );
+  if( !fieldExists( fieldName ))
+  {
+    // populate basic info
+    m_fields.emplace_back();
+    FieldDescription & field = m_fields.back();
+    field.name = fieldName;
+    field.location = location;
+    field.numComponents = components;
+    field.key = m_name + '_' + fieldName + "_dofIndex";
+    field.docstring = fieldName + " DoF indices";
+    // advanced processing
+    field.support = processedRegions;
+  }
+  else
+  {
+    FieldDescription & field = m_fields[getFieldIndex( fieldName )];
+    for( auto const & newRegion : processedRegions )
+    {
+      auto it = field.support.begin();
+      auto end1 = field.support.end();
```

Comment:
```suggestion
      auto const end1 = field.support.end();
```

Reply:
I think the conster is the better. You may even try to use `cend` 😉